lib/rules/no-useless-rename.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * @fileoverview Disallow renaming import, export, and destructured assignments to the same name.
+ * @author Kai Cataldo
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "disallow renaming import, export, and destructured assignments to the same name",
+            category: "ECMAScript 6",
+            recommended: false
+        },
+        fixable: "code",
+        schema: [
+            {
+                type: "object",
+                properties: {
+                    ignoreDestructuring: { type: "boolean" },
+                    ignoreImport: { type: "boolean" },
+                    ignoreExport: { type: "boolean" }
+                },
+                additionalProperties: false
+            }
+        ]
+    },
+
+    create: function(context) {
+        var options = context.options[0] || {},
+            ignoreDestructuring = options.ignoreDestructuring === true,
+            ignoreImport = options.ignoreImport === true,
+            ignoreExport = options.ignoreExport === true;
+
+        //--------------------------------------------------------------------------
+        // Helpers
+        //--------------------------------------------------------------------------
+
+        /**
+         * Reports error for unnecessarily renamed assignments
+         * @param {ASTNode} node - node to report
+         * @param {ASTNode} initial - node with initial name value
+         * @param {ASTNode} result - node with new name value
+         * @param {string} type - the type of the offending node
+         * @returns {void}
+         */
+        function reportError(node, initial, result, type) {
+            var name = initial.type === "Identifier" ? initial.name : initial.value;
+
+            return context.report({
+                node: node,
+                message: "{{type}} {{name}} unnecessarily renamed.",
+                data: {
+                    name: name,
+                    type: type
+                },
+                fix: function(fixer) {
+                    return fixer.replaceTextRange([
+                        initial.range[0],
+                        result.range[1]
+                    ], name);
```

Comment:
Replace the whole thing with just the name - this allows us to fix when keys are String Literals as well